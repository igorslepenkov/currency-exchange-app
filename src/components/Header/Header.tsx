import * as React from "react";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Divider from "@mui/material/Divider";
import Drawer from "@mui/material/Drawer";
import IconButton from "@mui/material/IconButton";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemText from "@mui/material/ListItemText";
import MenuIcon from "@mui/icons-material/Menu";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import Button from "@mui/material/Button";
import { ROUTE, RouteKey } from "../../router";
import { useNavigate } from "react-router-dom";
import { LinkWithoutStyles } from "../LinkWithoutStyles";

const drawerWidth = 240;

export const Header = () => {
  const [mobileOpen, setMobileOpen] = React.useState(false);

  const navigate = useNavigate();

  const routeKeys = Object.keys(ROUTE) as unknown as RouteKey[];

  const handleDrawerToggle = () => {
    setMobileOpen((prevState) => !prevState);
  };

  const drawer = (
    <Box onClick={handleDrawerToggle} sx={{ textAlign: "center" }}>
      <Typography variant="h4" sx={{ my: 2 }}>
        MUI
      </Typography>
      <Divider />
      <List>
        {routeKeys.map((key) => (
          <ListItem key={key} disablePadding>
            <ListItemButton
              sx={{ textAlign: "center" }}
              onClick={() => navigate(ROUTE[key])}
            >
              <ListItemText
                primary={key}
                primaryTypographyProps={{ variant: "h5" }}
              />
            </ListItemButton>
          </ListItem>
        ))}
      </List>
    </Box>
  );

  return (
    <>
      <AppBar
        component="header"
        position="static"
        sx={(theme) => ({ backgroundColor: theme.palette.background.paper })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            edge="start"
            onClick={handleDrawerToggle}
            sx={{ mr: 2, display: { sm: "none" } }}
          >
            <MenuIcon fontSize="large" />
          </IconButton>
          <Typography
            variant="h6"
            component="div"
            sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
          >
            <Typography variant="h5">EXHANGE APP</Typography>
          </Typography>
          <Box sx={{ display: { xs: "none", sm: "block" } }}>
            {routeKeys.map((key) => (
              <Button key={key} sx={{ color: "#fff" }}>
                <Typography variant="h6">
                  <LinkWithoutStyles to={ROUTE[key]}>{key}</LinkWithoutStyles>{" "}
                </Typography>
              </Button>
            ))}
          </Box>
        </Toolbar>
      </AppBar>
      <Box component="nav">
        <Drawer
          variant="temporary"
          open={mobileOpen}
          onClose={handleDrawerToggle}
          ModalProps={{
            keepMounted: true, // Better open performance on mobile.
          }}
          sx={{
            display: { xs: "block", sm: "none" },
            "& .MuiDrawer-paper": {
              boxSizing: "border-box",
              width: drawerWidth,
            },
          }}
        >
          {drawer}
        </Drawer>
      </Box>
    </>
  );
};
