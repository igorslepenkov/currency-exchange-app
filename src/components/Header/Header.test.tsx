import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Header } from "./Header";
import { render } from "@testing-library/react";

describe(Header, () => {
  it("Renders Header correctly", () => {
    const history = createMemoryHistory();

    const tree = render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );

    expect(tree).toMatchSnapshot();
  });
});
