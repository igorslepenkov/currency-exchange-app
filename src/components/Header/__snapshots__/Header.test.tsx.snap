// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders Header correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1061a19-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-6iz4mj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiTypography-root MuiTypography-h6 css-p7bec2-MuiTypography-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              EXHANGE APP
            </h5>
          </div>
          <div
            class="MuiBox-root css-130f8nx"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <a
                  class="sc-beqWaB gYiUh"
                  href="/"
                >
                  Home
                </a>
                 
              </h6>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <a
                  class="sc-beqWaB gYiUh"
                  href="/converter"
                >
                  Converter
                </a>
                 
              </h6>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                <a
                  class="sc-beqWaB gYiUh"
                  href="/currencies"
                >
                  Currencies
                </a>
                 
              </h6>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
      <nav
        class="MuiBox-root css-0"
      />
    </div>
    <div
      aria-hidden="true"
      class="MuiDrawer-root MuiDrawer-modal MuiModal-root MuiModal-hidden css-1pgoau0-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft css-4t3x6l-MuiPaper-root-MuiDrawer-paper"
        style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
        tabindex="-1"
      >
        <div
          class="MuiBox-root css-xi606m"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-ue75iy-MuiTypography-root"
          >
            MUI
          </h4>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1d4vqq0-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 MuiListItemText-primary css-1qvr50w-MuiTypography-root"
                  >
                    Home
                  </h5>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1d4vqq0-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 MuiListItemText-primary css-1qvr50w-MuiTypography-root"
                  >
                    Converter
                  </h5>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1d4vqq0-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 MuiListItemText-primary css-1qvr50w-MuiTypography-root"
                  >
                    Currencies
                  </h5>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1061a19-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-6iz4mj-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-h6 css-p7bec2-MuiTypography-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            EXHANGE APP
          </h5>
        </div>
        <div
          class="MuiBox-root css-130f8nx"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              <a
                class="sc-beqWaB gYiUh"
                href="/"
              >
                Home
              </a>
               
            </h6>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              <a
                class="sc-beqWaB gYiUh"
                href="/converter"
              >
                Converter
              </a>
               
            </h6>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              <a
                class="sc-beqWaB gYiUh"
                href="/currencies"
              >
                Currencies
              </a>
               
            </h6>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <nav
      class="MuiBox-root css-0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
